covid_tracking/app.py

```diff
@@ -76,8 +76,27 @@ def add_percapita_stats(df):
             df[c + '_pc'] = 1E6 * df[c] / df['population']
     return df
 
+def add_derived_metrics(df,endpoint):
+    """Adding metrics derived from the base stats:
+        newCases: new cases on this day for this state
+        posPerTest: positive test results/ total tests
+    :param df: dataFrame
+    """
+    df['posPerTest'] = df['positive']/df['total']
+
+    if 'states' in endpoint:
+        df['newCases'] = df.sort_values('date').groupby('state').positive.diff() if 'date' in df else df.groupby('state').positive.diff()
+    elif 'us' in endpoint:
```

Comment:
Can you make this operate on fields in the data instead (i.e. just look for 'state' and 'date' in the data frame)?  The substring 'us' works now but that's quite ambiguous and could match a lot of different endpoints that would shouldn't be matched here in the future.